README.md

```diff
@@ -45,6 +45,29 @@ To disable the feature add the following:
 add_filter( 'hm.smart-media.cropper', '__return_false' );
 ```
 
+The image cropping IU provides support for updating Gutenberg block attributes based on the current selection using the following function:
+
+```js
+window.SmartMedia.registerAttachmentToBlockAttributesMap( 'core/image', ( size, image, attachment ) => {
+  if ( ! image.label ) {
+    return null;
+  }
+  return {
+    sizeSlug: size,
+    url: image.url,
+    width: image.width,
+    height: image.height,
+  };
+} );
+```
+
+The function takes 2 parameters:
+
+- `block`: The name of the block to map attributes for
+- `callback`: A function that accepts the image `size` name, an `image` object containing url, width, height, crop data and label for the image size, and lastly the full attachment data object.
+
+The callback should return an object or `null`. Passing `null` will prevent updating the currently selected block.
```

Comment:
What's the shape of the returned object?

Reply:
It's variable, depends on the block's attributes. Not sure how best to show this really, it's such a gnarly system to interact with meaningfully. Maybe there's a better approach tbh, like a filter or something, forgot those were a thing in JS now.